cpp/ql/lib/experimental/semmle/code/cpp/semantic/analysis/RangeAnalysis.qll

```diff
@@ -542,12 +542,32 @@ private predicate unequalIntegralSsa(
 ) {
   exists(SemExpr e, int d1, int d2 |
     unequalFlowStepIntegralSsa(v, pos, e, d1, reason) and
-    bounded(e, b, d2, true, _, _, _) and
-    bounded(e, b, d2, false, _, _, _) and
+    boundedUpper(e, b, d1) and
+    boundedLower(e, b, d2) and
     delta = d2 + d1
   )
 }
 
+/**
+ * Holds if `b + delta` is an upper bound for `e`.
+ *
+ * This predicate only exists to prevent a bad standard order in `unequalIntegralSsa`.
+ */
+pragma[nomagic]
```

Comment:
Is `nomagic` sufficient, or do we need `noinline` too? I seem to remember that one implies the other, but I can't remember in which direction.

Reply:
`nomagic` implies `noinline`. So this should be correct as-is.